ambry-replication/src/main/java/com/github/ambry/replication/ReplicaThread.java

```diff
@@ -900,11 +911,14 @@ void processReplicaMetadataResponse(Set<MessageInfo> missingRemoteStoreMessages,
    * @param messageInfo message information of the blob from remote replica
    * @param remoteReplicaInfo remote replica information
    * @param localKey local blob information
+   * @param localMessageInfo message information of the blob from the local replica
    * @throws StoreException
    */
   public void applyUpdatesToBlobInLocalStore(MessageInfo messageInfo, RemoteReplicaInfo remoteReplicaInfo,
-      BlobId localKey) throws StoreException {
-    MessageInfo localMessageInfo = remoteReplicaInfo.getLocalStore().findKey(localKey);
+      BlobId localKey, MessageInfo localMessageInfo) throws StoreException {
+    if (localMessageInfo == null) {
```

Comment:
I would move this check to the calling method, and just do Objects.assertNonNull() here.